api/v1/logicalvolume_types.go

```diff
@@ -18,6 +18,17 @@ type LogicalVolumeSpec struct {
 	NodeName    string            `json:"nodeName"`
 	Size        resource.Quantity `json:"size"`
 	DeviceClass string            `json:"deviceClass,omitempty"`
+
+	// 'source' specifies the logicalvolume name of the source; if present.
+	// +kubebuilder:validation:Optional
+	Source string `json:"source"`
+
+	//'accessType' specifies how the user intends to consume the snapshot logical volume.
+	// Allowed values are: "ro" (read-only) and "rw" (read-write).
+	// Since the kubernetes snapshots are Read-Only, set accessType as 'ro' to activate thin-snapshots as read-only volumes.
+	// On the other hand, set accessType as 'rw' to activate thin-snapshots as read-write volumes.
+	// +kubebuilder:validation:Optional
+	AccessType string `json:"accessType"`
```

Comment:
This should also be an omitEmpty field and should not be present for a regular LogialVolume